internal/mesh/internal/types/http_route_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/internal/catalog"
+	"github.com/hashicorp/consul/internal/resource/resourcetest"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/consul/proto/private/prototest"
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+// TODO(rb): add mutation tests
+
+func TestValidateHTTPRoute(t *testing.T) {
+	type testcase struct {
+		route     *pbmesh.HTTPRoute
+		expectErr string
+	}
+
+	run := func(t *testing.T, tc testcase) {
+		res := resourcetest.Resource(HTTPRouteType, "api").
+			WithData(t, tc.route).
+			Build()
+
+		err := ValidateHTTPRoute(res)
+
+		// Verify that validate didn't actually change the object.
+		got := resourcetest.MustDecode[pbmesh.HTTPRoute, *pbmesh.HTTPRoute](t, res)
+		prototest.AssertDeepEqual(t, tc.route, got.Data)
+
+		if tc.expectErr == "" {
+			require.NoError(t, err)
+		} else {
+			testutil.RequireErrorContains(t, err, tc.expectErr)
+		}
+	}
+
+	cases := map[string]testcase{
```

Comment:
There's probably some missing cases here and I see the todos below. Did you want to add them later?

Reply:
That was what NET-5066 was for.